### PR TITLE
Added custom rounding increments for formatting distance

### LINKED ISSUE #628: Feature Request - NavigationView - Use smaller increments for rounding small distances
Small feature request.

After testing a bunch with the NavigationView, it seems that it always will round to 50 meters regardless of the distance left. This feels quite weird when the distance to the upcoming step is less than 100m. 

At first I thought about the idea to have 25 meter intervals, but in iOS it seems to be even lower increments. Maybe use the same setup with rounding and the intervals as in iOS?

Semi related: https://github.com/mapbox/mapbox-navigation-ios/issues/681

### PR DESCRIPTION
**Pull request by devotaaabel**
Fixes #628

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/DistanceFormatter.java [hunk 1]
```diff
@@ -48,9 +51,12 @@
    * @param context  from which to get localized strings from
    * @param language for which language
    * @param unitType to use, or NONE_SPECIFIED to use default for locale country
+   * @param roundingIncrement increment by which to round small distances
    */
   public DistanceFormatter(Context context, @NonNull String language,
-                           @NonNull @DirectionsCriteria.VoiceUnitCriteria String unitType) {
+                           @NonNull @DirectionsCriteria.VoiceUnitCriteria String unitType,
+                           @NavigationConstants.RoundingIncrement int roundingIncrement) {
```
Comment (danesfeder) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/DistanceFormatter.java]: Can we also update `DistanceFormatter#shouldUpdate` to consider `roundingIncrement` as well?

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/SummaryBottomSheet.java [hunk 2]
```diff
@@ -163,6 +167,15 @@ public void setTimeFormat(@NavigationTimeFormat.Type int type) {
     this.timeFormatType = type;
   }
 
+  /**
+   * Sets the rounding increment which will be used for rounding small distances
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/SummaryBottomSheet.java]: Let's be more specific with the javadoc here - maybe provide an example to illustrate.

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java [hunk 3]
```diff
@@ -361,14 +364,23 @@ public void setUnitType(@DirectionsCriteria.VoiceUnitCriteria String unitType) {
     checkForDistanceFormatterUpdate();
   }
 
+  /**
+   * Sets the rounding increment which will be used for rounding small distances
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java]: Same comment re: javadoc

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java [hunk 4]
```diff
@@ -528,11 +529,19 @@ private void initializeOnMoveListener() {
...
   private void establish(NavigationViewOptions options) {
     LocaleUtils localeUtils = new LocaleUtils();
+
+    setRoundingIncrement(options);
     establishLanguage(localeUtils, options);
     establishUnitType(localeUtils, options);
     establishTimeFormat(options);
   }
 
+  private void setRoundingIncrement(NavigationViewOptions navigationViewOptions) {
```
Comment (danesfeder) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: For uniformity here, might make sense to update `set` to `establish`

**Comment by devotaaabel** [comment]
@danesfeder ready for re-review

**Pull request merged**